src/api.cc

```diff
@@ -272,11 +272,13 @@ CheckInResult AkliteClient::CheckIn() const {
   LOG_INFO << "Searching for matching TUF Targets...";
   auto matchingTargets = filterTargets(tuf_repo_->GetTargets(), hw_id_, client_->tags, secondary_hwids_);
   if (matchingTargets.empty()) {
-    err_msg =
-        "Couldn't find Targets matching the current device tag and hardware ID; check a device tag or a hardware ID";
+    // TODO: consider reporting about it to the backend to make it easier to figure out
```

Comment:
That's a good idea. However, if a device tag is invalid then DG returns 404, so we'll get the checkin error much earlier than this spot.

The idea is still interesting, it will require changing both the client and the backend sides, so I would create a ticket to consider it.

Reply:
I understand, still idea is interesting, I am going to create ticket for it.